### ISSUE TITLE
Can't access ds02734 dataset through Datalad

### ISSUE THREAD

**Issue by soichih** [OpenNeuroOrg/openneuro#1429]
datalad get is failing with the following error message when I try to access dataset stored on ds001219.

```
get(error): /mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-14/anat/sub-14_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219]
staging 5dfecd1d32bff0635ce289d0
('datalad get', u'/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-14/anat/sub-14_T1w.nii.gz')
```

ref https://github.com/datalad/datalad/issues/3923

**Comment by soichih** [comment]
ds000158 is having a similar problem

get(error): /mnt/datalad/datasets.datalad.org/openneuro/ds000158/sub-100/func/sub-100_task-passivelistening_bold.nii.gz (file) [not available; Try making some of these repositories available:; 	063333d8-7918-4026-bbf0-f694a56afa7f -- root@6dedfdc60b41:/datalad/ds000158]
staging 5e1ae1b71875e11c6e94a8cd
('datalad get', u'/mnt/datalad/datasets.datalad.org/openneuro/ds000158/sub-100/func/sub-100_task-passivelistening_bold.nii.gz')

**Comment by franklin-feingold** [comment]
hey @soichih - it appears [ds001219](https://github.com/OpenNeuroDatasets/ds001219) and [000158](https://github.com/OpenNeuroDatasets/ds000158) have their github remotes, perhaps reattempting to access these datasets?

**Comment by soichih** [comment]
@franklin-feingold Do you mean I should try datalad get again? I just tried but I am still seeing the same error message

```
get(error): /mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/anat/sub-10_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219]
staging 5e1d23291875e161a294c603
('datalad get', u'/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/anat/sub-10_T1w.nii.gz')
```

Maybe I need to do datalad update first?

**Comment by soichih** [comment]
I am seeing a similar issue with ds001021 also.

**Comment by franklin-feingold** [comment]
perhaps updating the packages and datalad because the git remotes are there and it is pointing to the right spot - we have been making improvements in this

**Comment by soichih** [comment]
I've updated the datalad CLI to version 0.12.1, and also git pull / datalad update -r on all of our datasets.

 `datalad get` seems to work now on files that didn't work previously. :) Like all files under ds001021 seems to work.

```
brlife@wrangler:/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/EEG$ datalad get EEG_data_sub10_run_01.mat 
get(ok): sub-10/EEG/EEG_data_sub10_run_01.mat (file) [from s3-PUBLIC...]   

brlife@wrangler:/mnt/datalad/datasets.datalad.org/openneuro/ds001021/sub-A00008326/ses-DS2/func$ datalad get sub-A00008326_ses-DS2_task-rest_acq-CAP_bold.nii.gz
get(ok): sub-A00008326/ses-DS2/func/sub-A00008326_ses-DS2_task-rest_acq-CAP_bold.nii.gz (file) [from s3-PUBLIC...] 
```

However, I am still seeing the same error message on other files. like..

```
brlife@wrangler:/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/anat$ datalad get sub-10_T1w.nii.gz 
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219 [get(/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/anat/sub-10_T1w.nii.gz)] 
get(error): sub-10/anat/sub-10_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219]
```

git log shows that this dataset hasn't bee touched since last Jan.

```
brlife@wrangler:/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/anat$ git log -v
commit ed296b72787dbd573ba4791b1af989287f967315 (HEAD -> master, origin/synced/master, origin/master, origin/HEAD)
Author: Andrea Pisauro <andrea.pisauro@gmail.com>
Date:   Sat Jan 26 12:01:40 2019 +0000

    [DATALAD] added content

...

```

**Comment by franklin-feingold** [comment]
to clarify - ds001021 is all set now? Down to certain files in certain datasets may not be accessible?

I am trying to narrow down and clarify what maybe still having issues

**Comment by soichih** [comment]
Yes, ds001021 is all good. 

But not ds001219.
> datasets.datalad.org/openneuro/ds001219/sub-10/EEG/EEG_events_sub10.mat

```
datalad get EEG_events_sub10.mat 
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219 [get(/mnt/datalad/datasets.datalad.org/openneuro/ds001219/sub-10/EEG/EEG_events_sub10.mat)] 
get(error): sub-10/EEG/EEG_events_sub10.mat (file) [not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219]

```

**Comment by franklin-feingold** [comment]
It appears that is not the only file from ds001219? I saw an anat file in your code block. I am clarifying the unavailable files

**Comment by soichih** [comment]
Correct. anat file is broken also. I believe there are quite a few files that doesn't work when I do datalad get across different dataset.

**Comment by nellh** [comment]
I believe ds001219 is fixed now after a re-export. Please make sure you update your git-annex branch from the latest on GitHub and let us know if you are still having issues.

**Comment by soichih** [comment]
Thanks!

Could you explain how to "update your git-annex branch"?

I did the following but it doesn't seem to work..

brlife@wrangler:/mnt/datalad/datasets.datalad.org/openneuro/ds001219$ git
pull origin git-annex
From http://datasets.datalad.org/openneuro/ds001219/
 * branch            git-annex  -> FETCH_HEAD
fatal: refusing to merge unrelated histories


I also did "git pull && datalad update -r" but still no luck..

**Comment by nellh** [comment]
`datalad update` should work but alternatively you can `git fetch origin git-annex:git-annex` or checkout the git-annex branch and then pull it. The git-annex branch stores the git-annex metadata needed to resolve annexed objects in the master branch (or any other).

**Comment by soichih** [comment]
I tried `git fetch origin git-annex:git-annex` but still no luck. 

Here is a way to reproduce this problem.

```
$ git clone http://datasets.datalad.org/openneuro/ds001219/.git
$ cd ds001219
$ datalad get sub-10/anat/sub-10_T1w.nii.gz
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219 [get(/home/hayashis/test/datalad/ds001219/sub-10/anat/sub-10_T1w.nii.gz)] 
get(error): sub-10/anat/sub-10_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 	30982af9-1bcd-4f66-8e56-79cc3143b587 -- root@79cd556289f8:/datalad/ds001219]
```

**Comment by nellh** [comment]
I think the issue that is not the remote provided by OpenNeuro. The OpenNeuro git remote is https://github.com/OpenNeuroDatasets/ds001219.git

**Comment by soichih** [comment]
Ah.. I see. 

@yarikoptic Could you explain what this means? I thought datasets.datalad.org basically *mounts* openneuro repo (whichever that might be. ) If that's not what datalad superdataset does, should I git clone the http://github.com/OpenNeuroDatasets along with datalad superdataset on our end?

**Comment by yarikoptic** [comment]
There is no https://github.com/OpenNeuroDatasets superdataset. We created one at http://datasets.datalad.org/openneuro and host clones. We update them from time to time but not automatically yet. You could establish your own collection of clones straight from GitHub organization with some scripting.

**Comment by soichih** [comment]
@yarikoptic OK. I will filter out openneuro datasets on datasets.datalad.org and host the upstream OpenNeuroDatasets to pull openneuro datasets from. 

Is it difficult to setup the automated cloning from OpenNeuroDatasets?

**Comment by soichih** [comment]
I've made a script like this to clone all OpenNeuroDatasets.

```
#!/usr/bin/env 
const token = "--------------github api token----------------";
const rp = require('request-promise-native');
const cp = require('child_process');

async function load_repos() {
	let clone_urls = [];
	for(let page = 1;page < 3;++page) {
		let repos = await rp.get('https://api.github.com/orgs/OpenNeuroDatasets/repos?page='+page, {
			json: true, 
			headers: {'User-Agent': 'nodejs', Authorization: 'token '+token}
		});
		console.log(page, repos.length);
		repos.forEach(repo=>{
			console.log(repo.clone_url);
			clone_urls.push(repo.clone_url);
		});
		if(repos.length == 0) break;
	}
	return clone_urls;
}

load_repos().then(urls=>{
	urls.forEach(url=>{
		cp.execSync("datalad install -r "+url);
	});
});

```

**Comment by yarikoptic** [comment]
> Is it difficult to setup the automated cloning from OpenNeuroDatasets?

not at all.  My difficulty was/is that I do not just clone them, but also run our metadata extractors (so that users could use `datalad search`). For that I need to download all the data first. That is how I become the pain-in-the-... for everyone by reporting some of those issues I encounter with data downloads.  Then I run extractors (which also could fail for a variety of reasons), and come back with more pain... and then I aggregate that metadata and we are yet to fix datalad to make that painless, and only then I share.

But may be I should decouple those two steps (having clones, extract metadata) and then making clones updated regularly, while metadata - when I have time to go through the pains ;)  Will look into it whenever get a chance.

**Comment by soichih** [comment]
@yarikoptic Thanks for the explanation.. I understand your pain better now.. :cry: I think it's a good idea to decouple the cloning v.s. metadata extraction process. But I understand sometimes it's better to not expose features that don't work reliably yet. 

@nellh @franklin-feingold I've imported data from OpenNeuroDatasets and I can now successfully datalad-get those files that I was having problem previously. Please feel free to close this.

**Comment by soichih** [comment]
Sorry I just found another dataset that's still having this problem.. Could you re-open this?

> https://github.com/OpenNeuroDatasets/ds001966.git
```
brlife@wrangler:/mnt/datalad/OpenNeuroDatasets/ds001966/sub-06/anat$ datalad get sub-06_T1w.nii.gz
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 	d3a5395b-fd41-4c7d-a81d-29983c67cf88 -- root@a41ddf4d1478:/datalad/ds001966; ; (Note that these git remotes have annex-ignore set: origin) [get(/mnt/datalad/OpenNeuroDatasets/ds001966/sub-06/anat/sub-06_T1w.nii.gz)] 
get(error): sub-06/anat/sub-06_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 	d3a5395b-fd41-4c7d-a81d-29983c67cf88 -- root@a41ddf4d1478:/datalad/ds001966; ; (Note that these git remotes have annex-ignore set: origin)]
```

I *can* datalad get fine for subject 01-05, but not 06-29.

**Comment by franklin-feingold** [comment]
We found that ds1966 was not defaced so I am working with the dataset owner to bring it back online

**Comment by soichih** [comment]
@franklin-feingold 

I was randomly testing things and 2 out of 4 tests failed.

```
brlife@wrangler2:/mnt/datalad/OpenNeuroDatasets/ds002549/sub-006/ses-bhb/anat$ datalad get sub-006_ses-bhb_run-1_T1w.nii
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 34175de8-4e8f-42c4-a934-9940069558c5 -- s3-PUBLIC;   6c08b424-ce6a-4072-88cd-dcf952d26802 -- root@openneuro-prod-dataset-worker-8:/datalad/ds002549; ; (Note that these git remotes have annex-ignore set: origin) [get(/mnt/datalad/OpenNeuroDatasets/ds002549/sub-006/ses-bhb/anat/sub-006_ses-bhb_run-1_T1w.nii)]
get(error): sub-006/ses-bhb/anat/sub-006_ses-bhb_run-1_T1w.nii (file) [not available; Try making some of these repositories available:; 34175de8-4e8f-42c4-a934-9940069558c5 -- s3-PUBLIC
```

```
brlife@wrangler2:/mnt/datalad/OpenNeuroDatasets/ds002578/sub-001/anat$ datalad get sub-001_T1w.nii.gz
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 1e44b03b-428c-4d74-abb7-204559cea96d -- s3-PUBLIC;   dbfaead0-2991-4cd7-ad24-0443ea449d93 -- root@openneuro-prod-dataset-worker-4:/datalad/ds002578; ; (Note that these git remotes have annex-ignore set: origin) [get(/mnt/datalad/OpenNeuroDatasets/ds002578/sub-001/anat/sub-001_T1w.nii.gz)]
get(error): sub-001/anat/sub-001_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 1e44b03b-428c-4d74-abb7-204559cea96d -- s3-PUBLIC;   dbfaead0-2991-4cd7-ad24-0443ea449d93 -- root@openneuro-prod-dataset-worker-4:/datalad/ds002578; ; (Note that these git remotes have annex-ignore set: origin)]
```

It could be just some bad luck / temporary glitch.. but FYI.

@yarikoptic @mih 

I am also seeing an odd behavior from datalad.. it doesn't display anything but just quitting.

```
brlife@wrangler2:/mnt/datalad/OpenNeuroDatasets/ds000113/sub-01/ses-forrestgump/anat$ datalad get sub-01_ses-forrestgump_mod-T2w_defacemask.nii.gz
brlife@wrangler2:/mnt/datalad/OpenNeuroDatasets/ds000113/sub-01/ses-forrestgump/anat$ 

```

These are all happening on our backup data IO server for brainlife.io, so it could be a host issue.. but I don't know how to troubleshoot this.

**Comment by soichih** [comment]
Another random test failure. On ds002221.

```
[ERROR] not available; Try making some of these repositories available:; 1b454cc1-c5fc-437d-a165-e24297255653 -- s3-PUBLIC; a7972bdd-ab9c-4209-a110-1d8eda04a8fd -- root@openneuro-prod-dataset-worker-14:/datalad/ds002221; ; (Note that these git remotes have annex-ignore set: origin) [get(/mnt/datalad/OpenNeuroDatasets/ds002221/sub-11/func/sub-11_task-silentreading_run-2_bold.nii.gz)]
```

**Comment by franklin-feingold** [comment]
hey @soichih - sorry for my delayed response! 

I just checked on S3 and the remotes for: ds000113, ds002221, ds002549, and ds002578 - and confirmed they are there. These may be associated with some fixing of permissions on remotes 

I looked at datalad's [openneuro datasets](http://datasets.datalad.org/?dir=/openneuro) and did not notice the datasets there either - perhaps @yarikoptic may have some insights

**Comment by yarikoptic** [comment]
for the recent datasets -- see https://github.com/OpenNeuroOrg/datalad-service/issues/129

as for http://datasets.datalad.org/?dir=/openneuro -- the update on our end is not yet fully automated, I will update our clones there. It has nothing to do with inability to access data(sets) from https://github.com/OpenNeuroDatasets

**Comment by soichih** [comment]
@yarikoptic  

This fixed the issue for ds002221

```
git annex enableremote s3-PUBLIC
```

I can now datalad get!

However, it didn't work for ds000030, ds000030, ds000113, ds000217, ds000221, ds001417, ds002236

```
$ datalad install -r ds000113
[INFO] Installing <Dataset path=/mnt/datalad/OpenNeuroDatasets/ds000113> underneath /mnt/datalad/OpenNeuroDatasets/ds000113 recursively 
$ cd ds000113 && git annex enableremote s3-PUBLIC
git-annex: First run: git-annex init
$ git-annex init
init  
  Remote origin not usable by git-annex; setting annex-ignore
ok
(recording state in git...)
$ git annex enableremote s3-PUBLIC
git-annex: Unknown remote name.
	origin
(No special remotes are currently known; perhaps use initremote instead?)
```

**Comment by yarikoptic** [comment]
~~given the spread of datasets, probably different issues.~~
looking at ds000113 - seemed to have some issues -- https://github.com/OpenNeuroOrg/openneuro/issues/889 ... and that poor thing seems to not even have `git-annex` branch pushed.. the same for 217 

```shell
> git clone https://github.com/OpenNeuroDatasets/ds000113
Cloning into 'ds000113'...
remote: Enumerating objects: 6381, done.
remote: Total 6381 (delta 0), reused 0 (delta 0), pack-reused 6381
Receiving objects: 100% (6381/6381), 67.23 MiB | 6.33 MiB/s, done.
Resolving deltas: 100% (865/865), done.
Updating files: 100% (6409/6409), done.
git clone https://github.com/OpenNeuroDatasets/ds000113  7.17s user 1.10s system 63% cpu 13.081 total

> cd ds000113 

> git branch -a                       
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/master
```

the same for the others, filed https://github.com/OpenNeuroOrg/datalad-service/issues/130

**Comment by soichih** [comment]
@yarikoptic 

I found another one > OpenNeuroDatasets/ds001966

The magic of `git annex enableremote s3-PUBLIC` doesn't cure it..

```
brlife@wrangler2:/mnt/datalad/OpenNeuroDatasets/ds001966/sub-26/anat$ git annex enableremote s3-PUBLIC
enableremote s3-PUBLIC ok
(recording state in git...)
brlife@wrangler2:/mnt/datalad/OpenNeuroDatasets/ds001966/sub-26/anat$ datalad get sub-26_T1w.nii.gz 
[WARNING] Running get resulted in stderr output: git-annex: get: 1 failed
 
[ERROR  ] not available; Try making some of these repositories available:; 	d3a5395b-fd41-4c7d-a81d-29983c67cf88 -- root@a41ddf4d1478:/datalad/ds001966; ; (Note that these git remotes have annex-ignore set: origin) [get(/mnt/datalad/OpenNeuroDatasets/ds001966/sub-26/anat/sub-26_T1w.nii.gz)] 
get(error): sub-26/anat/sub-26_T1w.nii.gz (file) [not available; Try making some of these repositories available:; 	d3a5395b-fd41-4c7d-a81d-29983c67cf88 -- root@a41ddf4d1478:/datalad/ds001966; ; (Note that these git remotes have annex-ignore set: origin)]
```

Is this known issue also?

**Comment by franklin-feingold** [comment]
@soichih yep this dataset has been deleted

**Comment by soichih** [comment]
@franklin-feingold 

I didn't know that Openneuro datasets can be "deleted". When you say "delete", you mean it actually remove the data stored in s3, correct? Not just hide it from making it discoverable?

In my humble opinion, once a dataset is published, OpenNeuro should not allow users to remove the data. It can be made to *hide* from public view, for cases where the data is faulty or there is a new version, etc... Otherwise, people won't be able to reproduce results using data from OpenNeuro. Also, people could be downloading data prior to the deletion, so I think it really doesn't make sense to *delete* data once published..??

At the moment, I don't really have a good way of handling datasets that are "deleted" on brainlife.io. We could reproduce any generated data-derivatives, but not the raw data. So if OpenNeuro dataset could be deleted, then we could end up with publications (with derivatives) that there is really no way of reproducing. 

Do you know how often do people delete datasets on OpenNeuro?

Also, what is the best way to get notified/detect if OpenNeuro dataset is removed?

**Comment by soichih** [comment]
Ref > https://github.com/OpenNeuroOrg/openneuro/issues/1584

**Comment by franklin-feingold** [comment]
Perhaps I can clarify this - I define delete as not accessible and removed. A use case we are currently working with is handling non defaced datasets (related https://github.com/OpenNeuroOrg/openneuro/issues/1574). We cannot share non defaced datasets so in these cases the dataset has been deleted from the web version. We found that our processes may not have automated this deleted process (which gives evidence to why deleted datasets on the web UI are still on AWS). The dataset is reuploaded (with the defaced data). So the data is deleted and reuploaded (we are redirecting our links and DOI traffic to the new accession number). I have been keeping track of this mapping (https://github.com/OpenNeuroOrg/openneuro/issues/1176). 

In general, other than this defacing I have not seen too much dataset deletion after the dataset owner has made their set public. We are working on a facial recognition tool that can be deployed on OpenNeuro to resolve this.

There is not currently a process for notifications like that. I keep our [metadata sheet](metadata.openneuro.org) updated for what datasets are publicly available along with metadata information.

In the past this process for deleting was agreed upon until another approach presents itself. The ones that have already been deleted will still be removed from GitHub and AWS. Moving forward there can be a better practice

**Comment by franklin-feingold** [comment]
> Ref > #1584

regarding ds000229 is not empty but [unable to access remote](https://github.com/OpenNeuroOrg/openneuro/issues/1584#issuecomment-612187248), confirmed the dataset is on AWS, GitHub OpenNeurodatasets, and on datalad datasets

**Comment by soichih** [comment]
@franklin-feingold Thanks. I think it's reasonable that some dataset needs to be removed if the owner forgets to deface.. 

Just for curiosity, is there some datasets (outside of OpenNeuro) that participants have been informed and consented that their data will be made public without defacing? For such datasets, defacing should be optional? 

You could also handle this by just go-ahead and run defacing upon data upload to OpenNeuro - regardless of it's already defaced or not? Would it impact the data quality?

> There is not currently a process for notifications like that. I keep our metadata sheet updated for what datasets are publicly available along with metadata information.

I was able to access the metadata sheet link last week, but not anymore.. So I am thinking about doing something like this..

1) Iterate over github repo on https://github.com/OpenNeuroDatasets
2) Query https://openneuro.org/crn/graphql and see if the dataset exists
3) If it exists, import / update as normal
4) If it doesn't exist, act accordingly to on our end..

I need to think about what to do with data that's already processed, but based on your comment, this should be a rare occurrence. Would something like that work you think?

**Comment by franklin-feingold** [comment]
> Just for curiosity, is there some datasets (outside of OpenNeuro) that participants have been informed and consented that their data will be made public without defacing? For such datasets, defacing should be optional?

In this case, these data could be uploaded without defacing. I know of studies with fMRI-EEG taken simultaneously the facial features are important for analysis. It would be up to the repository and what their policies are. 

> You could also handle this by just go-ahead and run defacing upon data upload to OpenNeuro - regardless of it's already defaced or not? Would it impact the data quality?

We are working on a facial recognition tool (determine if there is a face) - this will run during the upload phase. Regarding impacting data quality - let me circle back to this (depending on the tool yes it can affect the image quality if you are try to deface a defaced image) 

Regarding metadata sheet, weird perhaps may you try and see if this works - metadata.openneuro.org? (the previous link was prepended with GitHub)

Regarding handling dataset existance - that could work! The graphql will output the status (exist or do not have access). As we shore up our delete processes the OpenNeuroDatasets collection will be much closer to what is available on OpenNeuro.

Yes, as we move forward this should be a rare occurrence. Perhaps an approach could be to freeze as historical data but then link to the updated dataset. I am recording [mappings of datasets](https://github.com/OpenNeuroOrg/openneuro/issues/1176) that fit into the delete -> reupload cases

**Comment by soichih** [comment]
@franklin-feingold 

I am now working on implementing a script to walk through all existing OpenNeuro dataset entries in our DB, and check to see if the dataset still exists by querying them through OpenNeuro's graphql API. 

I am running a query like this for each dataset.

```
query { 
dataset(id:\"${dataset_id}\") { id name public } 
}
```

I've noticed that this query is often very slow (especially when I run it for the first time for a given dataset, and it often times out with StatusCode 504. Is there something wrong with the way I am querying it?

**Comment by franklin-feingold** [comment]
hey @soichih 

sorry for the delayed response! Yesterday we spun up a [documentation site](https://docs.openneuro.org/api) (first pass - can be strengthened) that may be useful! 

The command looks good - not sure of optimization techniques (we are addressing timeouts through our current performance enhancements activities). Perhaps the auditing can be cornered to checking newer datasets? There are sets (like ds numbers <1000) that are from OpenfMRI. These are very unlikely to be removed from OpenNeuro. Another potential quick audit for dataset existence is validating against my [metadata sheet](https://docs.google.com/spreadsheets/u/1/d/1rsVlKg0vBzkx7XUGK4joky9cM8umtkQRpJ2Y-5d6x7c/edit#gid=1226202843/). I keep this updated with information associated with public datasets

**Comment by soichih** [comment]
@franklin-feingold Thanks for the API doc. I believe I was doing the right thing?

Anyway, I basically keep re-trying, and eventually I was able to check all datasets. What I've observed was that, for dataset IDs that I have not queried before, it was taking a very long time (and 504 timeout sometimes). Once I was able to query once, next time I ran the same query, it returns very quickly. So, by keep retrying, I eventually got to the bottom of the list. I check all datasets once a day, and so far I have not had the timeout issue anymore. If the GraphQL approach turns out to be insufficient, I will explore other alternatives - like looking at the metadata sheet.

Thank you for your help!
